internal/encryption/encryption_backend.go

```diff
@@ -0,0 +1,23 @@
+package encryption
+
+// EncryptionBackend defines the interface for encryption operations.
+// Implementations should handle the encryption and decryption of sensitive data.
+type EncryptionBackend interface {
+	Decrypt(ciphertext []byte) ([]byte, error)
+	Encrypt(plaintext []byte) ([]byte, error)
+}
+
+// NoEncryptionBackend is a no-op implementation of the EncryptionBackend interface.
+// It used when encryption backend is configured to none.
+type NoEncryptionBackend struct {
+}
+
+// Decrypt returns the ciphertext as is for NoEncryptionBackend.
+func (n NoEncryptionBackend) Decrypt(ciphertext []byte) ([]byte, error) {
+	return ciphertext, nil
+}
+
+// Encrypt returns the plaintext as is for NoEncryptionBackend.
+func (n NoEncryptionBackend) Encrypt(plaintext []byte) ([]byte, error) {
+	return plaintext, nil
+}
```

Comment:
As mentioned in my comment above, the "no encryption" content can belong in its own file, i.e. 1 file per backend.